clients/rospy/src/rospy/impl/tcpros_base.py

```diff
@@ -568,17 +571,27 @@ def connect(self, dest_addr, dest_port, endpoint_id, timeout=None):
             if not isinstance(e, socket.error):
                 # FATAL: no reconnection as error is unknown
                 self.close()
-            elif not isinstance(e, socket.timeout) and e.errno not in [100, 101, 102, 103, 110, 112, 113]:
+            elif not isinstance(e, socket.timeout) and e.errno not in [100, 101, 102, 103, 104, 110, 112, 113]:
                 # reconnect in follow cases, otherwise close the socket:
                 # 1. socket.timeout: on timeouts caused by delays on wireless links
                 # 2. ENETDOWN (100), ENETUNREACH (101), ENETRESET (102), ECONNABORTED (103):
                 #     while using ROS_HOSTNAME ros binds to a specific interface. Theses errors
                 #     are thrown on interface shutdown e.g. on reconnection in LTE networks
                 # 3. ETIMEDOUT (110): same like 1. (for completeness)
                 # 4. EHOSTDOWN (112), EHOSTUNREACH (113): while network and/or DNS-server is not reachable
+                # 5. ECONNRESET (104): The connection was reset by the peer. Connections may be backed up we
```

Comment:
I understand everything here except trying to reconnect on ECONNRESET (which would mean TCP sent a RST packet). The only normal ways this happens is if there is no one listening on the port at the remote host (and retrying would be pointless) or if the other side closed the connection on purpose (and retrying would also be pointless). Under what circumstance should we retry when getting a RST?

Reply:
I actually have a test that exposed this. If I connect to the ros tcp port with nc and just hold it open so that the accept loop is stuck. Then I spin up 30 rostopic echo commands and let them sit for a while some of them will eventually get ECONNRESET. Then when I release the nc command (or let the recv timeout do it's thing) the rostopic processes that got the ECONNRESET will not be connected to the publisher and never will and they will sit there not recieving any of the messages. So there was a case where ECONNRESET was sent when someone was listening on the socket and was continually for the whole time.

Reply:
I was forgetting that there are two error codes for a RST packet, ECONNRESET and ECONNREFUSED. We will get the later on a brand new connection, so this is OK